cocos/renderer/backend/Types.cpp

```diff
@@ -0,0 +1,16 @@
+#include "Types.h"
+
+
+CC_BACKEND_BEGIN
+
+
+bool UniformLocation::operator==(const UniformLocation &other) const
+{
+    return shaderStage == other.shaderStage && location == other.location;
+}
+
+std::size_t UniformLocation::operator()(const UniformLocation &uniform) const {
+    return (size_t)(shaderStage) || (size_t)(location & 0xFFFFFFF0);
+}
```

Comment:
What's the use case?

Reply:
This provides a the hash function for `UniformLocation`
in `ProgramState::_callbackUniforms`
```c++
    std::unordered_map<UniformLocation, UniformCallback, UniformLocation>   _callbackUniforms;
```

Reply:
why is `location & 0xFFFFFFF0`?

Reply:
It is a bug, I have fixed it